apps/api/src/app/environments-v1/usecases/output-renderers/hydrate-email-schema.usecase.ts

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable no-param-reassign */
+import { Injectable } from '@nestjs/common';
+import { TipTapNode } from '@novu/shared';
+import { z } from 'zod';
+import { MasterPayload } from '../construct-framework-workflow';
+import { HydrateEmailSchemaCommand } from './hydrate-email-schema.command';
+
+@Injectable()
+export class HydrateEmailSchemaUseCase {
```

Comment:
One thing that sounds a bit confusing is why do we have hydrate and expand email use cases separately. I'd expect a single use case that does both.

Reply:
I'm trying to create clear maintainable logic. there are two distinct behaviors here, we hydrate (from pointers to data) and than perform logic on data, those are hard to follow algorithms so I don't want to create a job security algorithem but one which is easy to learn, I'll add a comment explaining it